Source/Core/Core/PowerPC/Jit64/Jit_LoadStorePaired.cpp

```diff
@@ -305,9 +331,45 @@ void Jit64::psq_lXX(UGeckoInstruction inst)
 	gqr.offset += 2;
 
 	AND(32, R(RSCRATCH2), gqr);
-	MOVZX(32, 8, RSCRATCH, R(RSCRATCH2));
+	MOVZX(32, 8, RSCRATCH_EXTRA, R(RSCRATCH2));
+
+	BitSet32 registersInUse = CallerSavedRegistersInUse();
+	registersInUse[RSCRATCH2] = true;
+	registersInUse[RSCRATCH_EXTRA] = true;
+	if (w)
+	{
+		CMP(8, R(RSCRATCH2), Imm8(0));
+		FixupBranch check_4 = J_CC(CC_NE);
+		SafeLoadToReg(RSCRATCH, R(RSCRATCH), 32, 0, registersInUse, false);
+		FixupBranch end1 = J();
+		SetJumpTarget(check_4);
+		TEST(8, R(RSCRATCH2), Imm8(1));
+		FixupBranch check_2 = J_CC(CC_Z);
+		SafeLoadToReg(RSCRATCH, R(RSCRATCH), 16, 0, registersInUse, false);
+		FixupBranch end2 = J();
+		SetJumpTarget(check_2);
+		SafeLoadToReg(RSCRATCH, R(RSCRATCH), 8, 0, registersInUse, false);
+		SetJumpTarget(end1);
+		SetJumpTarget(end2);
+	}
+	else
+	{
+		CMP(8, R(RSCRATCH2), Imm8(0));
+		FixupBranch check_8 = J_CC(CC_NE);
+		SafeLoadToReg(RSCRATCH, R(RSCRATCH), 64, 0, registersInUse, false);
+		FixupBranch end1 = J();
+		SetJumpTarget(check_8);
+		TEST(8, R(RSCRATCH2), Imm8(1));
+		FixupBranch check_4 = J_CC(CC_Z);
+		SafeLoadToReg(RSCRATCH, R(RSCRATCH), 32, 0, registersInUse, false);
+		FixupBranch end2 = J();
+		SetJumpTarget(check_4);
+		SafeLoadToReg(RSCRATCH, R(RSCRATCH), 16, 0, registersInUse, false);
+		SetJumpTarget(end1);
+		SetJumpTarget(end2);
+	}
```

Comment:
I agree... but this seemed like the least terrible solution short of rewriting our paired loadstore implementation.